code/app/src/main/java/com/example/goldencarrot/data/model/user/UserImpl.java

```diff
@@ -9,17 +9,28 @@ public class UserImpl implements User{
     private Optional<String> phoneNumber;
     private String uId;
 
+    // Facility-specific fields
+    private Optional<String> facilityName; // Facility ID, if applicable
+    private Optional<String> location; // Facility location
+    private Optional<String> imageURL; // Facility image URL
+
     public UserImpl(){}
 
     public UserImpl(final String email,
                     final String userType,
                     final String name,
-                    final Optional<String> phoneNumber) throws Exception{
+                    final Optional<String> phoneNumber,
```

Comment:
sorry I guess I wasn't clear. 

the facilityClass should inherit the User methods not just add methods in the User

this is problematic because UserImpl could be an Organizer, Participant or Admin.

Reply:
Participant and admin users should not have a facility name

You could create another class called Organizer that inherits the UserImpl class

Please see this:
https://www.geeksforgeeks.org/inheritance-in-java/

If you do so, please make appropriate Unit tests for this class, and it should go under data/model/user

Reply:
Okay I see. I'll get those changes cleared. Thanks for the insight!